From: Sven Eckelmann <sven@narfation.org>
Date: Wed, 11 Jun 2025 19:41:23 +0200
Subject: locking/local_lock: Avoid shadowed variable 'l'

    /home/build_test/build_env/linux-build/linux/v6.15.1/include/linux/local_lock.h:88:1: warning: symbol 'l' shadows an earlier one
    /home/build_test/build_env/linux-build/linux/v6.15.1/include/linux/local_lock.h:88:1: originally declared here
    net/batman-adv/log.c: note: in included file (through /home/build_test/build_env/linux-build/linux/v6.15.1/include/linux/mmzone.h, /home/build_test/build_env/linux-build/linux/v6.15.1/include/linux/gfp.h, ...):

Fixes: 51339d99c013 ("locking/local_lock, mm: replace localtry_ helpers with local_trylock_t type")
Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/include/linux/local_lock_internal.h b/include/linux/local_lock_internal.h
index 8d5ac16a9b17941b3db8297b0c85383b2289eef1..8ce06c54eaecd30add01434c4f844b528220856f 100644
--- a/include/linux/local_lock_internal.h
+++ b/include/linux/local_lock_internal.h
@@ -97,17 +97,17 @@ do {								\
 #define __local_lock_acquire(lock)					\
 	do {								\
 		local_trylock_t *tl;					\
-		local_lock_t *l;					\
+		local_lock_t *la;					\
 									\
-		l = (local_lock_t *)this_cpu_ptr(lock);			\
-		tl = (local_trylock_t *)l;				\
+		la = (local_lock_t *)this_cpu_ptr(lock);			\
+		tl = (local_trylock_t *)la;				\
 		_Generic((lock),					\
 			__percpu local_trylock_t *: ({			\
 				lockdep_assert(tl->acquired == 0);	\
 				WRITE_ONCE(tl->acquired, 1);		\
 			}),						\
 			__percpu local_lock_t *: (void)0);		\
-		local_lock_acquire(l);					\
+		local_lock_acquire(la);					\
 	} while (0)
 
 #define __local_lock(lock)					\
